### DISCUSSION TITLE
Can we use @rpath instead of an absolute path as the install name of a dynamic lib?

### DISCUSSION THREAD

**Discussion by avegetablechicken** [Homebrew/discussions#511] [Everyday usage]
for example.
$ otool -L libgflags.dylib                                       
libgflags.dylib:
        /usr/local/opt/gflags/lib/libgflags.2.2.dylib (compatibility version 2.2.0, current version 2.2.2)
        /usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 400.9.4)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.200.5)

Since `/usr/local/opt/gflags/lib/libgflags.2.2.dylib` is an absolute path, i don't know how to make it portable, such as capsulating in a peoject. `install_name_tool` fails when i want to change the self-independent path (the first line).

So is there a way to replace the absolute path with @rpath, either when compiled or loaded?

**Comment by avegetablechicken** [comment]
it is ok to use the cmd `install_name_tool -id`..., but can we make it when compiling?

**Reply by areese** [reply]
`How are applications supposed to find libraries in /opt/local/lib and /opt/homebrew/lib ?` #1600 looks relevant.

**Reply by carlocab** [reply]
You can have a look at https://github.com/Homebrew/brew/blob/master/Library/Homebrew/extend/os/mac/keg_relocate.rb. If you change some of the methods here, you can fix it so that dylib references in any bottle you pour are prefixed with `@rpath`.

You'll probably need to populate the appropriate `LC_RPATH` entries with the correct prefixes for this to work.

**Reply by ipatch** [reply]
@avegetablechicken 

were ever able to make a portable `dylib` for the formula you were working with?
